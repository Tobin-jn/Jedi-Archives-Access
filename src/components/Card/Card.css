.card-container article {
  width: 200px;
  margin: 1rem;
  background: #192973;
  border-radius: 10px;
}

.card-header {
  display: grid;
  grid-template-columns: 1fr 60px;
  background: #000;
  border-top-right-radius: 10px;
  border-top-left-radius: 10px;
}

.favorite-toggle {
  /*background: #000;*/
  background: url(./images/rebel-red.svg);
  width: 40px;
  height: 40px;
  display: inline-block;
  margin: auto;
}

article p,
.card-header h2 {
  margin-left: 1rem;
}