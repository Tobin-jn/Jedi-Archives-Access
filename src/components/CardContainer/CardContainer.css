.card-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  width: 1200px;
  margin: auto;
}

@media (max-width: 1200px) {
  .card-container {
    grid-template-columns: 1fr 1fr 1fr;
    width: 900px;
  }
}

@media (max-width: 900px) {
  .card-container {
    grid-template-columns: 1fr 1fr;
    width: 600px;
  }
}

@media (max-width: 600px) {
  .card-container {
    grid-template-columns: 1fr;
    width: 300px;
  }
}