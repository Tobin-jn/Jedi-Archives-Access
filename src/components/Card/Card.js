import React from 'react';
import './Card.css';
import Images from '../../images.js'

const Card = ( {data} ) => {

  const stats = data.info.map((point) => {
    return <p key={Object.keys(point)[0]}>{Object.keys(point)[0]}: <span>{point[Object.keys( point)[0]]}</span></p>
  })

  return (
    <article className="card">
      <section className="front">
        <img className='portrait' src={Images[data.name]} alt={data.name} />
      </section>
      <section className="back">
        <div className='card-header'>
          <h2>{data.name}</h2>
          <div className='favorite-toggle'></div>
        </div>
        <div className="card-stats">
          {stats}
        </div>
      </section>
    </article>
  );
}

export default Card;