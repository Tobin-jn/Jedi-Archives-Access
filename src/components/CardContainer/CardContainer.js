import React from 'react';
import Card from '../Card/Card'
import './CardContainer.css'

const CardContainer = (props) => {

  let cards;

  if (props.selection === 'people') {
    cards = props.people.map( person => {
      return  <Card 
                data={person} 
                key={person.name} 
                toggleFavorite={props.toggleFavorite} 
              />
    });
  } else if (props.selection === 'vehicles') {
    cards = props.vehicles.map( vehicle => {
      return  <Card 
                data={vehicle} 
                key={vehicle.name} 
                toggleFavorite={props.toggleFavorite} 
              />
    })
  } else if (props.selection === 'planets') {
    cards = props.planets.map( planet => {
      return  <Card 
                data={planet} 
                key={planet.name} 
                toggleFavorite={props.toggleFavorite} 
              />
    });
  } else if (props.selection === 'favorites') {
    cards = props.favorites.map( favorite => {
      return  <Card 
                data={favorite} 
                key={favorite.name} 
                toggleFavorite={props.toggleFavorite} 
              />
    });
  }

  
  
  return (
    <div className="card-container">
      {cards}
    </div>
  );
}

export default CardContainer;