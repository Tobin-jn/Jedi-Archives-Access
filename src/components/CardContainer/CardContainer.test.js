import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';
import Card from '../Card/Card';

describe('CardContainer', () => {
  let wrapper;
  let mockPeople;
  let mockSelection;
  let mockToggleFavorite;
  let mockError;
  let mockLoading;
  
  beforeEach(() => {
    mockPeople = [{
      name: 'Luke',
      info: [
        {homeworld: 'Earch'},
        {language: 'English'},
        {species: 'Human'},
        {population: 'billions'}
      ]
    }]
    
    mockToggleFavorite = jest.fn();
    mockError = false;
    mockLoading = false;
    mockSelection = 'people'

    wrapper = shallow(<CardContainer 
                        data={mockPeople}
                        selection={mockSelection}
                        toggleFavorite={mockToggleFavorite}
                        error={mockError}
                        loading={mockLoading}
                        totalFavorites={3} 
                      />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render all the data cards', () => {
    expect(wrapper.find(Card).length).toEqual(1);
  })

  it('should render Loading if loading is true', () => {
    let mockLoading = true

    expect(wrapper).toMatchSnapshot()
  })
    it('should render Error if error is true', () => {
    let mockError = true

    expect(wrapper).toMatchSnapshot()
  })
})