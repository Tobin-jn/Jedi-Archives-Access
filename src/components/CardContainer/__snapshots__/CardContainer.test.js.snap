// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot 1`] = `
<div
  aria-label="card-container"
  className="card-container"
>
  <Card
    data={
      Object {
        "info": Array [
          Object {
            "homeworld": "Earch",
          },
          Object {
            "language": "English",
          },
          Object {
            "species": "Human",
          },
          Object {
            "population": "billions",
          },
        ],
        "name": "Luke",
      }
    }
    key="Luke"
    toggleFavorite={[MockFunction]}
  />
  <CardCenterpiece
    selection="people"
    totalFavorites={3}
  />
</div>
`;

exports[`CardContainer should render Error if error is true 1`] = `
<div
  aria-label="card-container"
  className="card-container"
>
  <Card
    data={
      Object {
        "info": Array [
          Object {
            "homeworld": "Earch",
          },
          Object {
            "language": "English",
          },
          Object {
            "species": "Human",
          },
          Object {
            "population": "billions",
          },
        ],
        "name": "Luke",
      }
    }
    key="Luke"
    toggleFavorite={[MockFunction]}
  />
  <CardCenterpiece
    selection="people"
    totalFavorites={3}
  />
</div>
`;

exports[`CardContainer should render Loading if loading is true 1`] = `
<div
  aria-label="card-container"
  className="card-container"
>
  <Card
    data={
      Object {
        "info": Array [
          Object {
            "homeworld": "Earch",
          },
          Object {
            "language": "English",
          },
          Object {
            "species": "Human",
          },
          Object {
            "population": "billions",
          },
        ],
        "name": "Luke",
      }
    }
    key="Luke"
    toggleFavorite={[MockFunction]}
  />
  <CardCenterpiece
    selection="people"
    totalFavorites={3}
  />
</div>
`;
